File: backend/src/main/kotlin/com/example/backend/models/EnRayonInventaire.kt
package com.example.backend.models

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault
import java.time.Instant
import java.time.LocalDateTime

@Entity
@Table(name = "en_rayon_inventaire")
class EnRayonInventaire {
@Id
  @GeneratedValue(strategy = GenerationType.IDENTITY) // Use IDENTITY for auto-increment
  @Column(name = "id", nullable = false)
  var id: Long? = null

  @Column(name = "inventaire_id")
  var inventaireId: Int? = null

  @Column(name = "en_rayon_id")
  var enRayonId: String? = null

  @Column(name = "employe_id")
  var employeId: Int? = null

  @Column(name = "quantite_rayon")
  var quantiteRayon: Int? = null

  @Column(name = "quantite_inventaire")
  var quantiteInventaire: Int? = null

  @Column(name = "date_debut")
  var dateDebut: LocalDateTime? = null

  @Column(name = "date_fin")
  var dateFin: LocalDateTime? = null

  @Column(name = "type", length = 100)
  var type: String? = null

  @Column(name = "statut", length = 100)
  var statut: String? = null

  @ColumnDefault("0")
  @Column(name = "supprimer")
  var supprimer: Int? = null
}

File: backend_py/pharmaPython/app/models/en_rayon_inventaire.py
from sqlalchemy import Column, Integer, String, DateTime
from app.api.v1.db import Base

class EnRayonInventaire(Base):
  __tablename__ = "en_rayon_inventaire"
  __table_args__ = {"extend_existing": True}

  id = Column(Integer, primary_key=True, autoincrement=True)
  inventaire_id = Column(Integer)
  en_rayon_id = Column(String)
  employe_id = Column(Integer)
  quantite_rayon = Column(Integer)
  quantite_inventaire = Column(Integer)
  date_debut = Column(DateTime)
  date_fin = Column(DateTime)
  type = Column(String(100))
  statut = Column(String(100))
  supprimer = Column(Integer, default=0)
